### DISCUSSION TITLE
expo-splash-screen on iOS sometimes does not "hideAsync"

### DISCUSSION THREAD

**Discussion by artola** [expo/expo#28175] [Archived: Questions]
I found a bug on **expo-splash-screen** (v0.26.4) running on iOS 17.4.1 (iPhone 15 Pro Max).
When running the **RELEASE** version, I execute `preventAutoHideAsync` before mounting the App, and later on when the router `isReady` it triggers `hideAsync`.
But it not able to hide the Splash and it remains there if I do not retry.
The typings show that `hideAsync(): Promise<boolean>`, but it returns the first time `undefined`.
I do retry every 100ms (a couple of times) and returns `true`.
This is only happening on **RELEASE**, but it works on **Debug** or Simulator.

**Comment by brunowilliang** [comment]
hey @artola did you manage to solve it? i have the same problem

**Comment by AjaxVM** [comment]
We have been encountering this as well, after adding a retry I see it failed with undefined 3 times in our app, before succeeding as well. @Brunowilliang you could try doing the same, just look for `result === undefined` from `hideAsync` call.

**Comment by brentvatne** [comment]
this is a common symptom of other some other problem - usually something else has gone wrong while starting up your app, and the error is not handled properly, and the result is that the splash screen is never hidden because execution never reaches your hide splash screen call.

> [Runtime errors](https://docs.expo.dev/build-reference/troubleshooting/#runtime-errors)
>
> Common questions that fall under this category are: "my app runs well locally but crashes immediately when I run a build" or **"my app works in Expo Go but hangs on the splash screen in my build"**. When your app builds successfully but crashes or hangs when you run it, this is considered a runtime error.
>
> Refer to the ["Production errors" section of the debugging guide](https://docs.expo.dev/debugging/runtime-issues#production-errors) to learn how to locate logs when your release builds are crashing at runtime.
>
> If you can't find any useful information through this approach, try [narrowing down the source of the crash step by step](https://expo.fyi/manual-debugging).

**Reply by jkaufman** [reply]
Thank you, @brentvatne for that added context. We don't see any errors in console or our APM tool, but may be overlooking something. I'm still suspecting an unrelated bug involving the callbacks.

Edit: But we must be encountering something different since it reproduces in development builds, too.

**Comment by outaTiME** [comment]
I'm also having a problem with the 'expo-splash-screen' but it's different and it's only on Android, the splash screen hides itself automatically as if "ignoring" the `preventAutoHideAsync`.
More detailed information can be found in the issue: https://github.com/expo/expo/issues/30643
